### PR TITLE
Added differentiation between available and active toolbar

### PR DESCRIPTION
**Pull request by wachterjohannes**
This PR enables sulu to set the current toolbar or a specific user. Will be used in https://github.com/sulu/sulu/pull/2404

### CODE REVIEW

File: husky_extensions/ckeditor-extension.js [hunk 1]
```diff
@@ -127,9 +128,30 @@
...
+                        for (var section in toolbar) {
+                            var index = -1;
+
+                            while (index++ < toolbar[section].length) {
+                                if (!relations[toolbar[section][index]]) {
+                                    continue;
+                                }
+
+                                toolbar[section].splice.apply(
+                                    toolbar[section],
```
Comment (danrot) [inline husky_extensions/ckeditor-extension.js]: I that necessary?

File: husky_extensions/ckeditor-extension.js [hunk 2]
```diff
@@ -127,9 +128,30 @@
...
+                                    [index + 1, 0].concat(relations[toolbar[section][index]])
+                                );
```
Comment (danrot) [inline husky_extensions/ckeditor-extension.js]: This function call in general looks quite complicated to me. Isn't it possible to make that easier?

File: dist/husky.js [hunk 3]
```diff
@@ -47598,17 +47598,18 @@ define('__component__$url-input@husky',['services/husky/url-validator'], functio
 
             initialize: function(app) {
 
-                var toolbar = {
+                var availableToolbar = {
```
Comment (marcelmoosbrugger) [inline dist/husky.js]: Maybe "availableToolbars" or "availbleToolbarSettings" is a better name.

File: dist/husky.js [hunk 4]
```diff
@@ -47619,9 +47620,30 @@ define('__component__$url-input@husky',['services/husky/url-validator'], functio
...
                         BulletedList: 'list',
-                        PasteFromWord: 'file-word-o',
+                        PasteFromWord: 'clipboard',
                         Styles: 'header',
                         Source: 'code'
+                    },
+                    relations = {
+                        Link: ['Unlink']
+                    },
+                    expandToolbar = function(toolbar) {
```
Comment (marcelmoosbrugger) [inline dist/husky.js]: Please provide a comment, on what the function does. The function is not that easy.

File: husky_extensions/ckeditor-extension.js [hunk 5]
```diff
@@ -127,9 +128,40 @@
...
                         Source: 'code'
+                    },
+                    relations = {
+                        Link: ['Unlink']
+                    },
+
+                    /**
+                     * Expands a given toolbar. Iterates over all sections and
+                     * all section items in the toolbar. If there is a relation
+                     * defined for a section item it as all items of the relations
```
Comment (danrot) [inline husky_extensions/ckeditor-extension.js]: `it as`? Should probably be `it adds`.

File: husky_extensions/ckeditor-extension.js [hunk 6]
```diff
@@ -127,9 +128,40 @@
...
+                     * to that section.
+                     *
+                     * @param {Object} toolbar The toolbar to exapnd
```
Comment (danrot) [inline husky_extensions/ckeditor-extension.js]: `expand`

File: husky_extensions/ckeditor-extension.js [hunk 7]
```diff
@@ -140,20 +172,36 @@
...
-                    addToolbarButton: function(toolbarName, button, icon) {
-                        if (!toolbar[toolbarName]) {
-                            toolbar[toolbarName] = [];
+                    addToolbarButton: function(toolbarName, button, icon, buttonRelations) {
+                        if (!availableToolbarSections[toolbarName]) {
+                            availableToolbarSections[toolbarName] = [];
                         }
 
-                        toolbar[toolbarName].push(button);
+                        availableToolbarSections[toolbarName].push(button);
```
Comment (danrot) [inline husky_extensions/ckeditor-extension.js]: `availableToolbarSections[toolbarName]` is a bit confusing... does this object now contain toolbars or toolbar sections?

File: husky_extensions/ckeditor-extension.js [hunk 8]
```diff
@@ -140,20 +172,36 @@
...
 
                         if (!!icon) {
                             icons[button] = icon;
                         }
+
+                        if (!!relations) {
```
Comment (danrot) [inline husky_extensions/ckeditor-extension.js]: What is this check for? This variable always exists, doesn't it? And if there would be the possibility that it doesn't exist, just ignoring is probably not the best solution, is it?

File: husky_extensions/ckeditor-extension.js (hunk 7 above)
Comment (marcelmoosbrugger) [inline husky_extensions/ckeditor-extension.js]: `availableToolbarSections` essentially contains toolbar sections which can be combined to represent a toolbar. I renamed the `toolbarName` parameter to `sectionName`.

**Pull request merged**